### PR TITLE
refactor: wrap `gson` and `messagingEnabled` in factory funcs

### PR DESCRIPTION
**Pull request by sis0k0**
The above vars shouldn't be evaluated at build time, because the mksnapshot tool will fail (android runtime is not loaded, yet). That's why we need to wrap them in factory functions and instantiate them at runtime.

### CODE REVIEW

**Comment by sis0k0** [comment]
ping @EddyVerbruggen @PanayotCankov

**Comment by EddyVerbruggen** [comment]
I take your word for it! Thank you 👍

**Comment by EddyVerbruggen** [comment]
I'll release a new version tomorrow.

File: firebase.android.js [hunk 1]
```diff
@@ -15,13 +15,28 @@ var fbCallbackManager = null;
...
+function lazy(action) {
+    var _value;
+    var that = this;
+
+    return function() {
+        return that._value || (that._value = action());
+    }
+}
+
+var gson = lazy(function() {
```
Comment (EddyVerbruggen) [inline firebase.android.js]: Turned out I had to add `new` here (and 4 lines below) otherwise the cached values would interfere.

**Pull request merged**